app/home/routes.py

```diff
@@ -120,14 +123,30 @@ def user_listing(user_id):
 @blueprint.route("/property/update/<int:property_id>", methods=["GET", "POST"])
 @login_required
 def update_property(property_id):
-    from app.tasks import update_prop_images
+    from app.tasks import delete_img_obj
+    prop_to_update = Property.query.get(property_id)
 
     form = UpdatePropertyForm()
     if request.method == "POST" and form.validate_on_submit():
         if request.files["prop_photos"].filename:
-            img_files = request.files.getlist("prop_photos")
-            temp_dir, image_file_list = save_images_to_temp_folder(img_files)
-            update_prop_images(temp_dir, image_file_list, property_id)
+            if request.files["prop_photos"].filename:
+                # delete previous images before
+                delete_img_obj.delay(
+                    bucket,
+                    s3_prop_img_dir + prop_to_update.image_folder,
+                    json.loads(prop_to_update.photos)
+                )
+
+                img_files = request.files.getlist("prop_photos")
+                redis_img_keys = save_to_redis(img_files)
+
+                s3_upload_dir = generate_dir_name()
+                output_filenames = resize_images(redis_img_keys, s3_upload_dir)
+                img_list_to_json = json.dumps(output_filenames)
+
+                prop_to_update.image_folder = output_filenames[0]
+                prop_to_update.photos = img_list_to_json
+                db.session.commit()
```

Comment:
I  was under the impression that  a celery  worker is doing these for  both create  and update

Reply:
The celery worker just handles the resizing of images and uploads them to S3.